denops/@ddc-sources/shell_native.ts

```diff
@@ -34,7 +35,21 @@ export class Source extends BaseSource<Params> {
       await this.#printError(denops, `Invalid param: envs`);
       return;
     }
-    if (await fn.executable(denops, shell) !== 1) {
+
+    // Merge environment variables and localte the shell executable
+    const vimEnv = await fn.environ(denops) as Record<string, string>;
+    const mergedEnv = { ...vimEnv, ...envs };
+    const shellCommand = await accumulate(
+      denops,
+      async (denops) =>
+        (await Promise.all([
+          vars.environment.set(denops, "PATH", mergedEnv.PATH),
+          fn.exepath(denops, shell),
+          vars.environment.set(denops, "PATH", vimEnv.PATH),
+        ]))[1],
+    );
```

Comment:
Why the code is needed?

Please explain the code.

Reply:
If the user specified `params.envs.PATH`, `executable()` couldn't verify the command's existence, so I handled it that way.

However, I've reconsidered and changed the implementation to catch errors from `Deno.Command.spawn()` instead of pre checking.